client/state/ui/layout-focus/reducer.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import includes from 'lodash/includes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_FOCUS_SET,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_NEXT_FOCUS_SET,
+} from 'state/action-types';
+
+const initialState = { current: null, previous: null, next: null };
+const validAreas = [ 'content', 'sidebar', 'sites', 'preview' ];
+const isValidArea = area => includes( validAreas, area );
+
+export default function layoutFocus( state = initialState, action ) {
+	switch ( action.type ) {
+		case LAYOUT_FOCUS_SET:
+			if ( action.area === state.current || ! isValidArea( action.area ) ) {
```

Comment:
>So, throw an error here? Or just allow any layoutfocus to be set and invalid ones will just fail to do anything/break stuff?

Don't think we need to throw, just don't like the idea of accommodating setting an invalid area (at least in that it's handled gracefully by this condition). Rather, I think we should technically allow for the invalid area to be assigned, insofar that it will obviously not work as expected and potentially cause unexpected behavior. But we should be diligent about ensuring that we're not doing that 😄